OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs

```diff
@@ -86,7 +86,13 @@ public MainMenuLogic(Widget widget, World world, ModData modData)
 
 			mainMenu.Get<ButtonWidget>("EXTRAS_BUTTON").OnClick = () => SwitchMenu(MenuType.Extras);
 
-			mainMenu.Get<ButtonWidget>("QUIT_BUTTON").OnClick = Game.Exit;
+			mainMenu.Get<ButtonWidget>("QUIT_BUTTON").OnClick = () => SwitchMenu(MenuType.Quit);
```

Comment:
Can we keep this as-is? Compared to a keyboard press I don't think clicking the quit button will be accidental, so this just adds an extra step without benefit imo.

Reply:
unfortunately quit button is connected with escape key.
This changed line is basically gate to feature (yeah, some may argue) I added. It would not work anymore.

But I see your advice, I'll check possible solution to this, but this is my first experience with yaml, so I may not find it :)

Reply:
You can use the `OnKeyPress` handler:
```csharp
mainMenu.Get<ButtonWidget>("QUIT_BUTTON").OnClick = Game.Exit;
mainMenu.Get<ButtonWidget>("QUIT_BUTTON").OnKeyPress = (_) => SwitchMenu(MenuType.Quit);
```